tests/ui/float_cmp.rs

```diff
@@ -77,6 +77,13 @@ fn main() {
 
     assert_eq!(a, b); // no errors
 
+    let a1: [f32; 1] = [0.0];
+    let a2: [f32; 1] = [1.1];
+
+    assert_eq!(a1[0], a2[0]);
```

Comment:
If we want to do it right, this should not trigger the lint, since everything in `a1` ist constant `0.0`, so a comparison is ok here.

---

On a more general note, this enhancement needs more tests like the base case of comparing floats. (Comparing arrays of constant 0.0, constant {NEG_}INFINITY, ...)